### PR TITLE
Bunny conjse 1802 - Migrate to release-tools/publish-rubygem

### PR DESCRIPTION
**Pull request by bunnyhopkinton**
https://ca-il-jira.il.cyber-ark.com:8443/browse/CONJSE-1802

### CODE REVIEW

File: publish.sh [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
 
-docker pull registry.tld/conjurinc/publish-rubygem
+set -e
+echo "==> Starting publish.sh"
 
+#echo "==> Removing Gemfile.lock"
+#rm -f Gemfile.lock
+
+echo "==> Cloning Release-tools"
+git clone git@github.com:conjurinc/release-tools.git
```
Comment (neil-k-zero) [inline publish.sh:10]: Not sure if this should be the Atlantis repo?

File: Jenkinsfile [hunk 2]
```diff
@@ -44,18 +45,27 @@ pipeline {
...
-        deleteDir()
-      }
-    }
-  }
+        sh 'docker run -i --rm -v $PWD:/src -w /src --entrypoint /bin/sh alpine/git \
+            -c "git config --global --add safe.directory /src && \
+            git clean -fxd" '
+
+        
+        } //script
```
Comment (neil-k-zero) [inline Jenkinsfile]: Do we want the comments?

File: Jenkinsfile [hunk 3]
```diff
@@ -19,8 +19,9 @@ pipeline {
 
     // Only publish to RubyGems if branch is 'master'
     // AND someone confirms this stage within 5 minutes
-    stage('Publish to RubyGems?') {
 
+    stage('Publish to RubyGems') {
+/*
```
Comment (neil-k-zero) [inline Jenkinsfile]: coment?

File: memtar.gemspec [hunk 4]
```diff
@@ -12,14 +12,17 @@ Gem::Specification.new do |spec|
   spec.homepage      = "https://github.com/conjurinc/memtar"
   spec.license       = "MIT"
 
-  spec.files         = `git ls-files -z`.split("\x0")
+  #spec.files         = `git ls-files -z`.split("\x0")
```
Comment (neil-k-zero) [inline memtar.gemspec]: do we still need the comment

File: test.sh [hunk 5]
```diff
@@ -1,4 +1,15 @@
...
 
-docker build -f Dockerfile.test -t memtar_test .
-docker run --rm -v "$(pwd):/src" memtar_test rake spec
+echo "==> Starting test.sh"
+
+echo "==> Removing Gemfile.lock"
+rm -f Gemfile.lock
+
+echo "==> Docker Run"
+docker run --rm \
```
Comment (neil-k-zero) [inline test.sh:9]: do we want this to default to true?

**Review by neil-k-zero — COMMENTED** [review]
The build is still failing. Not sure how we plan to migrate this to Atlantis

File: publish.sh (hunk 1 above)
Comment (bunnyhopkinton) [reply publish.sh:10]: Cannot be done

File: Jenkinsfile (hunk 2 above)
Comment (bunnyhopkinton) [reply Jenkinsfile]: removed comments

File: memtar.gemspec (hunk 4 above)
Comment (bunnyhopkinton) [reply memtar.gemspec]: removed

File: Jenkinsfile (hunk 3 above)
Comment (bunnyhopkinton) [reply Jenkinsfile]: I forgot to uncomment this after testing. It's the when clause to make it only publish when a release is needed.

File: test.sh (hunk 5 above)
Comment (bunnyhopkinton) [reply test.sh:9]: @neil-k-zero , I'm not actually sure why it's set to default to true.

Comment (bunnyhopkinton) [reply test.sh:9]: Removed || :

File: memtar.gemspec [hunk 6]
```diff
@@ -12,14 +12,16 @@ Gem::Specification.new do |spec|
   spec.homepage      = "https://github.com/conjurinc/memtar"
   spec.license       = "MIT"
 
-  spec.files         = `git ls-files -z`.split("\x0")
+  spec.files          = `git ls-files`.split($\)
```
Comment (neil-k-zero) [inline memtar.gemspec]: Formatting is slightly off if you want to fix during a commit squash.

**Review by neil-k-zero — COMMENTED** [review]
Looks good, just needs a commit squash.

File: memtar.gemspec (hunk 6 above)
Comment (bunnyhopkinton) [reply memtar.gemspec]: Fixed!

**Comment by bunnyhopkinton** [comment]
> Looks good, just needs a commit squash.

Squashed!

**Comment by bunnyhopkinton** [comment]
> Looks good, just needs a commit squash.

Squashed!

**Review by neil-k-zero — APPROVED** [review]

**Pull request merged**